De-duplicate feature flags from Settings.yaml that are provided by UnifiedWebPreferences.yaml

brentfulgham opened the pull request:
#### 6eb6b7e29314edc08edee2733bb39e89b27d31c4
<pre>
De-duplicate feature flags from Settings.yaml that are provided by UnifiedWebPreferences.yaml
<a href="https://bugs.webkit.org/show_bug.cgi?id=250778">https://bugs.webkit.org/show_bug.cgi?id=250778</a>
&lt;rdar://problem/104389479&gt;

Reviewed by Elliott Williams.

A number of preferences defined in UnifiedWebPreferences.yaml are duplicated in Settings.yaml.
The generator handles this case properly, but it&apos;s confusing to developers.

This patch also removed the unused MediaDeviceIdentifierStorageDirectory item.

It also restores the &apos;Settings.yaml&apos; file to the Xcode project so that search will
locate items within it.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml: Clean up duplicates.
* Source/WebCore/WebCore.xcodeproj/project.pbxproj: Add missing Settings.yaml file.
* Source/WebCore/editing/cocoa/DataDetectorType.h: Add a &apos;None&apos; case.
* Source/WebCore/page/Settings.yaml: Remove duplicates.
* Source/WebKitLegacy/mac/Scripts/PreferencesTemplates/WebPreferencesDefinitions.h.erb: Downcast enums to integral types.

Canonical link: <a href="https://commits.webkit.org/259101@main">https://commits.webkit.org/259101@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/033302471d6306f80bc1cda4cb69749fe9ef3aa6

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/103814 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/12932 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/36758 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/113041 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/173351 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/107764 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/13955 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/3827 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/96063 "Built successfully") | [💥 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/112152 "An unexpected error occured. Recent messages:Failed to print configuration") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/109587 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/10763 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/93824 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/38479 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/92594 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/25432 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/80132 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/93928 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/6288 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/26819 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/90384 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/4056 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/6456 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/3358 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/29843 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/12440 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/46342 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/98993 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6266 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/8224 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/24913 "Passed tests") | 
| | | | | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on Source/WebCore/WebCore.xcodeproj/project.pbxproj:
```diff
@@ -12162,6 +12162,7 @@
 		7A6D3C7F28A4053D00C5C6EC /* NotificationResources.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; path = NotificationResources.h; sourceTree = "<group>"; };
 		7A6D3C8128A4055300C5C6EC /* NotificationResourcesLoader.cpp */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.cpp.cpp; path = NotificationResourcesLoader.cpp; sourceTree = "<group>"; };
 		7A6D3C8328A4056300C5C6EC /* NotificationResourcesLoader.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; path = NotificationResourcesLoader.h; sourceTree = "<group>"; };
+		7A7946642978815C00750A1B /* Settings.yaml */ = {isa = PBXFileReference; lastKnownFileType = text.yaml; path = Settings.yaml; sourceTree = "<group>"; };
```
Thank you for fixing this!

emw-apple commented on Source/WebCore/page/Settings.yaml:
```diff
@@ -253,12 +232,6 @@ ImagesEnabled:
     WebCore:
       default: true
 
-IsAccessibilityIsolatedTreeEnabled:
-  type: bool
-  defaultValue:
-    WebCore:
-      default: false
-
```
The declaration in UWP.yaml has `webcoreBinding: DeprecatedGlobalSettings`. Does it need to be moved to a per-instance setting there?

brentfulgham replied:
Good point. I'll double-check that.

brentfulgham replied:
Lots of code seems to expect it to live in `DeprecatedGlobalSettings`, and don't necessarily have a page context to grab the setting from. I think this is correct as-is.

brentfulgham committed:
De-duplicate feature flags from Settings.yaml that are provided by UnifiedWebPreferences.yaml
https://bugs.webkit.org/show_bug.cgi?id=250778
<rdar://problem/104389479>

Reviewed by Elliott Williams.

A number of preferences defined in UnifiedWebPreferences.yaml are duplicated in Settings.yaml.
The generator handles this case properly, but it's confusing to developers.

This patch also removed the unused MediaDeviceIdentifierStorageDirectory item.

It also restores the 'Settings.yaml' file to the Xcode project so that search will
locate items within it.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml: Clean up duplicates.
* Source/WebCore/WebCore.xcodeproj/project.pbxproj: Add missing Settings.yaml file.
* Source/WebCore/editing/cocoa/DataDetectorType.h: Add a 'None' case.
* Source/WebCore/page/Settings.yaml: Remove duplicates.
* Source/WebKitLegacy/mac/Scripts/PreferencesTemplates/WebPreferencesDefinitions.h.erb: Downcast enums to integral types.

Canonical link: https://commits.webkit.org/259101@main
